src/lang/langHelpers.ts

```diff
@@ -67,13 +70,14 @@ export async function executeAst({
 }> {
   try {
     if (!useFakeExecutor) {
-      engineCommandManager.endSession()
       // eslint-disable-next-line @typescript-eslint/no-floating-promises
-      engineCommandManager.startNewSession()
+      // this is going to be weird
+      // engineCommandManager.startNewSession()
```

Comment:
this is probabbly bad cc @Irev-Dev im unsure how to solve, maybe i make the check cache function seperate from the execute function so only optionally do we clear everything in the world? startNewSession appears to clear the artifact map or something and seems super important, so if it is i can split in two functions, if not let me know

Reply:
or i could trigfger something from the rust wasm side, wait thats a good call i can do that! as a post clear hook, we good, we good i got it ignore me!